MoltenVK/MoltenVK/Commands/MVKCommand.h

```diff
@@ -67,6 +67,10 @@ class MVKCommand : public MVKBaseObject, public MVKLinkableMixin<MVKCommand> {
 	/** Encodes this command on the specified command encoder. */
 	virtual void encode(MVKCommandEncoder* cmdEncoder) = 0;
 
+    virtual void encodeToICB(id<MTLIndirectCommandBuffer> icb) {
```

Comment:
Something like this could be useful, but I believe we'd need to have an additional context object provided with it so individual commands could look up relevant index/vertex/other buffers.